pkg/karmadactl/cmdinit/utils/util_test.go

```diff
@@ -1,26 +1,112 @@
 package utils
 
 import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
 	"os"
+	"path/filepath"
 	"testing"
 )
 
-const (
-	filePath = "./crds.tar.gz"
-	folder   = "./crds"
-)
+func init() {
+	output = ioutil.Discard
+}
 
 // TestDownloadFile test DownloadFile
+//nolint:gocyclo
 func TestDownloadFile(t *testing.T) {
-	defer os.Remove(filePath)
-	defer os.RemoveAll(folder)
+	const testFileName = "test.txt"
+	testFileContent := []byte("test")
+
+	serverTar := func() io.Reader {
+		tmpDir, err := os.MkdirTemp("", "karmada-test-download-server")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			_ = os.RemoveAll(tmpDir)
+		}()
+
+		// prepare test file
+		file := filepath.Join(tmpDir, testFileName)
+		if err = ioutil.WriteFile(file, testFileContent, 0600); err != nil {
+			t.Fatal(err)
+		}
```

Comment:
Maybe we don't need really create the file, we can just create a tarball, here is an example:
https://github.com/golang/go/blob/939f9fd64a740ef770a0ca3588f8c193dceebe83/src/archive/tar/example_test.go#L19

Then, I guess we can remove the `nolint:gocyclo` then.

Reply:
Just a reminder, the `ioutil.WriteFile` has been deprecated.

Reply:
Thanks, I will try it.